src/celeritas/optical/model/RayleighMfpCalculator.hh

```diff
@@ -0,0 +1,156 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/model/RayleighMfpCalculator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "celeritas/Constants.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/grid/GenericCalculator.hh"
+#include "celeritas/optical/detail/OpticalUtils.hh"
+
+#include "../MaterialView.hh"
+
+namespace celeritas
+{
+class MaterialParams;
+struct ImportOpticalRayleigh;
+
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Material properties used to calculate Rayleigh mean free paths.
+ */
+struct OpticalRayleighMaterial
+{
+    real_type scale_factor{1};  //!< User scale for the scattering length
+    real_type compressibility{0};  //!< Isothermal compressibility
+    real_type temperature{0};  //!< Material temperature
+
+    operator bool() const
+    {
+        return scale_factor > 0 && compressibility > 0 && temperature > 0;
+    }
+
+    static std::vector<OpticalRayleighMaterial>
+    from_import(std::vector<ImportOpticalRayleigh> const& imported,
+                ::celeritas::MaterialParams const& mat);
```

Comment:
This is unusual to have in a "runtime calculation" file. Since we don't have a `RayleighData.hh` file and that might be overkill, I think it's fine to keep the `OpticalRayleighMaterial` struct here. But could you make the `from_import` part of the model, and then add a convenience function to get the rayleigh data from the model for a given optical material ID?

Reply:
To clarify, by "runtime calculation" do you mean not setup time? I think this calculator/struct is only ever used during setup for the mfp construction and only inside the Rayleigh model, right? What would the convenience function be for?

Reply:
I was thinking ahead toward if we wanted to generate cross sections (mfps) on the fly, we could use this calculator. I assumed the `OpticalRayleighMaterial` might be used for testing? (If not, no worries)